### ISSUE TITLE
Frequently logged out from exercism

### ISSUE THREAD

**Issue by SaschaMann** [exercism/exercism#6104]
I'm frequently logged out of the website. This does not happen on GitHub or any other services I use with GitHub SSO. I'd estimate at least once a month.

Using GitHub SSO, latest Firefox, Win10.

**Comment by SaschaMann** [comment]
Notes from Jeremy on Slack:

> So for cookies, we're using entirely the Rails and Devise (our auth lib's) defaults. My suspicion would be that others sites more aggressively try to keep you logged in and constantly set cookies as that's valuable for them, and we don't. But there may also be "bad" defaults that I'm not aware of. The fact that this happens to both of you in Firefox might also suggest that that is more aggressive on cookie expiration than other browsers too. I haven't been logged out in Chrome since launch for example.

> It might be that Rails doesn't have a default expiration field (so they never expire by default) but that Firefox then takes any cookie that never expires and adds a date (e.g. 2 weeks or a month) on it. I'd need to read the source of Rails to check as the documentation doesn't specify the default value.

**Comment by SaschaMann** [comment]
Firefox shows me a 2 week expiration date for the  `remember_user_token` cookie, which might be responsible here?

**Comment by SaschaMann** [comment]
Without knowing any Rails, this might be what needs to be changed: https://github.com/exercism/website/blob/3de6029c02b0362ad9281e078b75c966f342330b/config/initializers/devise.rb#L198-L200

**Comment by SleeplessByte** [comment]
https://github.com/heartcombo/devise/blob/8593801130f2df94a50863b5db535c272b00efe1/lib/devise.rb#L122-L124

The default _is_ actually 2 weeks.

**Comment by iHiD** [comment]
@SleeplessByte Am I reading `extend_remember_period` correctly, that that extends the token every time a new session is created?

**Comment by SleeplessByte** [comment]
https://github.com/heartcombo/devise/blob/8593801130f2df94a50863b5db535c272b00efe1/lib/devise/strategies/rememberable.rb#L21-L30

Yes

**Comment by iHiD** [comment]
Yeah, that was how I read that too. I just wasn't sure exactly when that was called. Cool. I'll probably extend the default and enable this then.
